lib/elixir/lib/io.ex

```diff
@@ -416,7 +416,8 @@ defmodule IO do
   """
   @spec inspect(device, item, keyword) :: item when item: var
   def inspect(device, item, opts) when is_list(opts) do
-    label = if label = opts[:label], do: [to_chardata(label), ": "], else: []
+    {label, opts} = Keyword.pop(opts, :label)
+    label = if label, do: [to_chardata(label), ": "], else: []
     opts = Inspect.Opts.new(opts)
```

Comment:
changing Opts.new to use `struct!` could be considered a breaking change since before this, incorrect key names were silently ignored.